gis/grass7/Portfile

```diff
@@ -176,6 +177,50 @@ variant mysql5 description {Add MySQL 5 support} {
     configure.env-append    MYSQLD_CONFIG=${prefix}/lib/mysql5/bin/mysql_config
 }
 
+variant app requires gui description "Build application bundle to launch grass${main_version}" {
+    global app_name
+    set app_name_version [join [lrange [split ${version} "."] 0 1] "."]
+    set app_name GRASS-${app_name_version}.app
+
+    post-build {
+        xinstall -d -m 0755 ${worksrcpath}/${app_name}/Contents
+        xinstall -d -m 0755 ${worksrcpath}/${app_name}/Contents/Resources
+        xinstall -d -m 0755 ${worksrcpath}/${app_name}/Contents/MacOS
+
+        set script [open "${worksrcpath}/${app_name}/Contents/MacOS/GRASS" w 0755]
+        puts ${script} "#!/usr/bin/osascript"
+        puts ${script} ""
+        puts ${script} "tell application \"Terminal\" to do script \"${prefix}/bin/grass${main_version} --gui; exit\""
+        close ${script}
+
+        copy ${worksrcpath}/macosx/app/AppIcon.icns \
+            ${worksrcpath}/${app_name}/Contents/Resources/AppIcon.icns
+        copy ${worksrcpath}/macosx/app/GRASSDocument_gxw.icns \
+            ${worksrcpath}/${app_name}/Contents/Resources/GRASSDocument_gxw.icns
+
+        set info_plist ${worksrcpath}/${app_name}/Contents/Info.plist
+        copy ${worksrcpath}/macosx/app/Info.plist.in ${info_plist}
+
+        set version_file        [open ${worksrcpath}/include/VERSION]
+        set version_data        [split [read ${version_file}] "\n"]
+        close ${version_file}
+        set grass_version_major [lindex ${version_data} 0]
+        set grass_version_minor [lindex ${version_data} 1]
+        set grass_version_date  [lindex ${version_data} 3]
+        set bundle_version      [join [lrange ${version_data} 0 2] "."]
+
+        reinplace "s|@GRASS_VERSION_MAJOR@|${grass_version_major}|" ${info_plist}
+        reinplace "s|@GRASS_VERSION_MINOR@|${grass_version_minor}|" ${info_plist}
+        reinplace "s|@GRASS_VERSION_DATE@|${grass_version_date}|" ${info_plist}
+        reinplace "s|@BUNDLE_VERSION@|${bundle_version}|" ${info_plist}
+        reinplace "s|@DEPLOYMENT_TARGET@|${macosx_deployment_target}|" ${info_plist}
+    }
+
+    post-destroot {
+        copy ${worksrcpath}/${app_name} ${destroot}${applications_dir}
+    }
+}
```

Comment:
Why is this a variant? Just always run this code.

Reply:
GRASS GIS may be used only as a library (e.g. for gdal-grass and qgis3), and there may be power users not interested in the gui at all. Therefore it is useful to have +app as a variant. I can imagine it as a default though!

Reply:
Then they don't have to use the GUI. We prefer to build in everything where feasible. Especially as this does not add any dependencies.

Reply:
Well, it does add dependency.+app requires +gui, which in turn depends on the wxWidgets port.

Having that said, if you think the added dependency is reasonable I have no problem going all-in for gui and app.

Reply:
So what is the point of a `+gui` variant if it does not add an app? So how about adding this code to the existing `+gui `variant?

Reply:
Up to this point, the gui had to be started through terminal.

However, good point: adding the app bundle stuff to the gui variant makes sense.